extension/app/ts/AddressBook.tsx

```diff
@@ -0,0 +1,121 @@
+import { useEffect, useRef, useState } from 'preact/hooks'
+import { addressString } from './utils/bigint.js'
+import { AddressBookEntries } from './utils/user-interface-types.js'
+import Blockie from './components/subcomponents/PreactBlocky.js'
+import { GetAddressBookDataReply, MessageToPopup } from './utils/interceptor-messages.js'
+
+type ActiveFilter = 'My Active Addresses' | 'My Contacts' | 'Tokens' | 'Non Fungible Tokens' | 'Other Contracts'
+
+export function FilterLink(param: { name: ActiveFilter, currentFilter: ActiveFilter, setActiveFilter: (activeFilter: ActiveFilter) => void}) {
+	return <a
+		class = { param.currentFilter === param.name ? `is-active` : '' }
+		onClick = { () => param.setActiveFilter(param.name) }>
+			{ param.name }
+	</a>
+}
+
+export function AddressList(param: { addressBookEntries: AddressBookEntries | undefined }) {
```

Comment:
Consider destructuring here so you don't have to do `param.` everywhere.
```suggestion
export function AddressList({addressBookEntries}: { addressBookEntries: AddressBookEntries | undefined }) {
```